x-pack/plugins/alerting/server/plugin.ts

```diff
@@ -571,7 +566,7 @@ export class AlertingPlugin {
       uiSettings: core.uiSettings,
     });
 
-    const getRulesClientWithRequest = (request: KibanaRequest) => {
+    const getRulesClientWithRequest = async (request: KibanaRequest) => {
```

Comment:
Because the rules client is dependent on the alerting authorization class the way it is instantiated is changed. Check the "Instantiation of the alerting authorization class" section in my description. I can provide more details if you like. I was torn a lot with the decision but in the end, I decided to go with the refactor as the lifecycle of the classes and their dependencies are more clear IMO. Let me know what you think. I am curious!